### PR TITLE
-Generated buildings now strictly obey height limit

### PR DESCRIPTION
**Pull request by questmas**
-Fix bug where uncentered buildings would be rotated into weird shapes

### CODE REVIEW

File: Blender/modules/GenerateBuilding.py [hunk 1]
```diff
@@ -352,19 +355,21 @@ def generate_building_base(bm,
                         scale_face(bm, face, sx, 1, sz)
 
                     # Do a bit of rotation
-                    if random() > 0:
+                    if random() > 0.8:
```
Comment (NanyiJiang) [inline Blender/modules/GenerateBuilding.py]: lets make this a parameter we can tune(how 'curvy' a building is)

File: Blender/modules/GenerateBuilding.py [hunk 2]
```diff
@@ -329,19 +329,22 @@ def generate_building_base(bm,
...
+                    if random() > 0.75 and heightLeft > 0.0:
+                        nextHeight = hull_segment_length * 0.25 if hull_segment_length * 0.25 < heightLeft else heightLeft
                         face = extrude_face(
-                            bm, face, hull_segment_length * 0.25)
+                            bm, face, nextHeight)
+                        heightLeft -= nextHeight
 
+                    is_last_hull_segment = heightLeft <= 0.1
                     # Maybe apply some scaling
                     if random() > 0.5:
```
Comment (NanyiJiang) [inline Blender/modules/GenerateBuilding.py]: same here

File: Blender/modules/GenerateBuilding.py [hunk 3]
```diff
@@ -329,19 +329,22 @@ def generate_building_base(bm,
...
-            hull_segment_range = range(num_hull_segments)
-            for i in hull_segment_range:
-                is_last_hull_segment = i == hull_segment_range[-1]
-                val = random()
-                if val > 0.2:
+            while heightLeft > 0.1:
+                if random() > 0.2:
                     # Most of the time, extrude out the face with some random deviations
-                    face = extrude_face(bm, face, hull_segment_length)
-                    if random() > 0.75:
```
Comment (NanyiJiang) [inline Blender/modules/GenerateBuilding.py]: same here

File: Blender/modules/GenerateBuilding.py (hunk 1 above)
Comment (questmas) [reply Blender/modules/GenerateBuilding.py]: Done

File: Blender/modules/GenerateBuilding.py (hunk 2 above)
Comment (questmas) [reply Blender/modules/GenerateBuilding.py]: Done

File: Blender/modules/GenerateBuilding.py (hunk 3 above)
Comment (questmas) [reply Blender/modules/GenerateBuilding.py]: Done

**Pull request merged**